packages/transaction-controller/src/helpers/EtherscanRemoteTransactionSource.ts

```diff
@@ -51,20 +55,41 @@ export class EtherscanRemoteTransactionSource
   async fetchTransactions(
     request: RemoteTransactionSourceRequest,
   ): Promise<TransactionMeta[]> {
+    const releaseLock = await this.#mutex.acquire();
+    const acquiredTime = Date.now();
+
     const etherscanRequest: EtherscanTransactionRequest = {
       ...request,
       chainId: request.currentChainId,
     };
 
-    const transactions = this.#isTokenRequestPending
-      ? await this.#fetchTokenTransactions(request, etherscanRequest)
-      : await this.#fetchNormalTransactions(request, etherscanRequest);
+    try {
+      const transactions = this.#isTokenRequestPending
+        ? await this.#fetchTokenTransactions(request, etherscanRequest)
+        : await this.#fetchNormalTransactions(request, etherscanRequest);
+
+      if (this.#includeTokenTransfers) {
+        this.#isTokenRequestPending = !this.#isTokenRequestPending;
+      }
 
-    if (this.#includeTokenTransfers) {
-      this.#isTokenRequestPending = !this.#isTokenRequestPending;
+      return transactions;
+    } finally {
+      this.#releaseLockAfterInterval(acquiredTime, releaseLock);
     }
+  }
 
-    return transactions;
+  #releaseLockAfterInterval(acquireTime: number, releaseLock: () => void) {
```

Comment:
Nit: It seems like EtherscanRemoteTransactionSource might be easier to reason about and test if the sleep were added to the thing that uses this class instead of here, but we can make that change later.

Reply:
The intent of the `EtherscanRemoteTransactionSource` is to encapsulate the Etherscan specific domain,  requirements, and logic to retrieve transaction data.

As the rate limiting is specific to Etherscan, it seems ideal that this file owns that requirement and enforces it internally, so it's always respected regardless of the caller.